packages/react-form/src/hooks/submit.ts

```diff
@@ -1,40 +1,49 @@
-import {useState, useCallback, useRef} from 'react';
-import {useMountedRef} from '@shopify/react-hooks';
+import {useState, useCallback} from 'react';
+import {useLazyRef, useMountedRef} from '@shopify/react-hooks';
 
 import {
   FormMapping,
   SubmitHandler,
   SubmitResult,
   FieldBag,
   FormError,
+  DynamicListBag,
 } from '../types';
 import {
   propagateErrors,
   validateAll,
   getValues,
   makeCleanFields,
+  makeCleanDynamicList,
 } from '../utilities';
 
 export function useSubmit<T extends FieldBag>(
   onSubmit: SubmitHandler<FormMapping<T, 'value'>> = noopSubmission,
   fieldBag: T,
   makeCleanAfterSubmit = false,
+  dynamicListBag?: DynamicListBag,
```

Comment:
The code is very confusing now, `fieldBag` might or might not include the dynamic lists (see `fieldsWithLists` in form.ts). And in parallel we still pass `dynamicListBag`, I think we should refactor that. 

Maybe we should simply pass `makeCleanDynamicList(dynamicLists)` has a parameter?

Reply:
I am not sure I understand 😕 . I dont think the code is confusing. Note that `useSubmit` can be used on its own. In that case, we would like to give people the opportunity to pass in a dynamic list to clean their field

Reply:
My problem is: `fieldBag` already contains `dynamicListBag` (see fieldsWithLists in form.ts) and then we re-pass it a second time. That seems not very clean.

Reply:
Ideally it will have been nice if we didnt have to pass in the dynamic list and if cleanFields was able to clean the dynamic list without us having to pass it in but I think this is perhaps one of the drawbacks to not having the whole dynamicList in the FieldBag. It has to be passed into every function that will need to perform an operation on the list itself and not just the fields. Wdyt?

Reply:
fieldBag does not contain the DynamicListBag. It only contains fields that are in the dynamic list. Hence, we have to pass the dynamicList.

Reply:
Can we make a new issue for this and as I agree it would be awesome not to have to pass around certain values to everything (in this case `dynamicListBag`).